app/uk/gov/hmrc/agentsubscription/service/SubscriptionService.scala

```diff
@@ -128,7 +140,15 @@ class SubscriptionService @Inject() (
           safeId <- desConnector.createOverseasBusinessPartnerRecord(OverseasRegistrationRequest(application))
           _ <- agentOverseasApplicationConnector.updateApplicationStatus(ApplicationStatus.Registered, userId, Some(safeId))
           arnOpt <- subscribeAndEnrolOverseas(authIds, safeId, application.amlsDetails, application.agencyDetails)
-        } yield arnOpt
+        } yield {
+          val auditJson = Json.toJson(OverseasSubscriptionAuditDetail(
+            arnOpt,
+            safeId, application.agencyDetails.agencyName,
+            application.agencyDetails.agencyEmail, application.agencyDetails.agencyAddress, application.amlsDetails)).as[JsObject]
+
+          auditService.auditEvent(AgentSubscriptionEvent.OverseasAgentSubscription, "Overseas agent subscription", auditJson)
+          arnOpt
```

Comment:
we could move all the audit details construction into AuditService , but not a big deal now